File: AndroidStudioProjects/lab04/app/src/main/java/pbs/edu/lab04/MainActivity.java
package pbs.edu.lab04;

import android.content.Intent;
import android.os.Bundle;
import android.view.KeyEvent;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnLinear = findViewById(R.id.btnLinear);
        Button btnRelative = findViewById(R.id.btnRelative);

        btnLinear.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, SimpleListsActivity.class);
            startActivity(intent);
        });

        btnRelative.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, HarderListsActivity.class);
            startActivity(intent);
        });
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_ESCAPE) {
            finish();
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }
}

File: AndroidStudioProjects/lab05/app/src/main/java/pbs/edu/lab05/MainActivity.kt
package pbs.edu.lab05

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.KeyEvent
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.btnLinear).setOnClickListener {
            startActivity(Intent(this, SimpleListsActivity::class.java))
        }

        findViewById<Button>(R.id.btnRelative).setOnClickListener {
            startActivity(Intent(this, HarderListsActivity::class.java))
        }

        findViewById<Button>(R.id.btnTable).setOnClickListener {
            startActivity(Intent(this, ListsActivity::class.java))
        }
    }

    // obsługa klawisza ESC (na emulatorze / fizycznej klawiaturze)
    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
        if (keyCode == KeyEvent.KEYCODE_ESCAPE) {
            // przykładowe zachowanie: cofnięcie do poprzedniej aktywności
            onBackPressed()
            return true
        }
        return super.onKeyDown(keyCode, event)
    }
}
